packages/app-project/pages/_app.js

```diff
@@ -26,67 +28,110 @@ enableStaticRendering(typeof window === 'undefined')
 */
 function useStore(initialState) {
   const isServer = typeof window === 'undefined'
-  const store = useMemo(() => initStore(isServer, initialState), [isServer, initialState])
+  const store = useMemo(
+    () => initStore(isServer, initialState),
+    [isServer, initialState]
+  )
   return store
 }
 
 function MyApp({ Component, pageProps }) {
+  /* Handle the theme mode */
+  const [themeMode, setThemeMode] = useState('light')
+
+  useEffect(() => {
+    const isBrowser = typeof window !== 'undefined'
+    const localStorage = isBrowser ? window.localStorage : null
+
+    if (isBrowser && !localStorage?.getItem('theme')) {
+      if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+        setThemeMode('dark')
+        localStorage?.setItem('theme', 'dark')
+      }
+    } else if (isBrowser) {
+      setThemeMode(localStorage?.getItem('theme'))
+    }
+  }, [])
+
+  function toggleTheme() {
+    const newTheme = themeMode === 'light' ? 'dark' : 'light'
+
+    setThemeMode(newTheme)
+    localStorage?.setItem('theme', newTheme)
```

Comment:
Thanks for checking! I think it works as is since `toggleTheme()` is never called by a component on the server, only in the browser. Nor is it called by any tests so `localStorage` is always defined. However, for clarity I'm going to move the check for `isBrowser` and define `localStorage` in the MyApp scope here and in app-content-pages for consistency.

Reply:
Oh! And in app-root too